ast-checks/src/tree_utils.rs

```diff
@@ -0,0 +1,46 @@
+use rnix::{
+    types::*, NodeOrToken, SyntaxElement, SyntaxKind, SyntaxKind::*, SyntaxNode, WalkEvent,
+};
+use std::iter::{once, successors};
+
+pub fn walk_kind(node: &SyntaxNode, kind: SyntaxKind) -> impl Iterator<Item = SyntaxElement> {
+    node.preorder_with_tokens()
+        .filter_map(move |event| match event {
+            WalkEvent::Enter(element) => Some(element).filter(|it| it.kind() == kind),
+            WalkEvent::Leave(_) => None,
+        })
+}
+
+pub fn walk_keyvalues_filter_key<'a>(
+    node: &SyntaxNode,
+    key: &'a str,
+) -> impl Iterator<Item = KeyValue> + 'a {
+    walk_kind(node, NODE_KEY_VALUE).filter_map(move |element| match element {
+        NodeOrToken::Node(node) => {
+            let kv = KeyValue::cast(node).unwrap();
+            if kv.key().and_then(|e| Some(e.node().to_string())) == Some(key.to_string()) {
+                Some(kv)
+            } else {
+                None
+            }
+        }
+        NodeOrToken::Token(_tok) => {
+            unreachable!(
+                "walk_kind is iterating over kind=`NODE_KEY_VALUE`, which
+                          should only yield `SyntaxNode`s rather than `SytnaxToken`s"
+            );
+        }
+    })
+}
```

Comment:
I think this can be a `.filter_map(|element| element.into_token()).map(|token| ....)`

Reply:
Done